airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -178,6 +240,64 @@ def filter_records_newer_than_state(self, stream_state: Mapping[str, Any] = None
             yield from records_slice
 
 
+class IncrementalShopifyStreamWithDeletedEvents(IncrementalShopifyStream):
+    @property
+    @abstractmethod
+    def deleted_events_api_name(self) -> str:
+        """
+        The string value of the Shopify Events Object to pull:
+
+            articles -> Article
+            blogs -> Blog
+            custom_collections -> Collection
+            orders -> Order
+            pages -> Page
+            price_rules -> PriceRule
+            products -> Product
+
+        """
+
+    @property
+    def deleted_events(self) -> ShopifyDeletedEventsStream:
+        """
+        The Events stream instance to fetch the `destroyed` records for specified `deleted_events_api_name`, like: `Product`.
+        See more in `ShopifyDeletedEventsStream` class.
+        """
+        return ShopifyDeletedEventsStream(self.config, self.deleted_events_api_name)
+
+    @property
+    def default_deleted_state_comparison_value(self) -> Union[int, str]:
+        """
+        Set the default STATE comparison value for cases when the deleted record doesn't have it's value.
+        We expect the `deleted_at` cursor field for destroyed records would be always type of String.
+        """
+        return ""
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        We extend the stream state with `deleted` property to store the `destroyed` records STATE separetely from the Stream State.
+        """
+        state = super().get_updated_state(current_stream_state, latest_record)
```

Comment:
is it safer to say something like: 

```
if not latest_record.get("deleted_at"): 
  return  super().get_updated_state()
else: 
  <return state with deletion> 
```
? 

the reason is to avoid updating the state for non-deletions with a deleted record

Reply:
I don't imagine the scenario when the state would be updated following the deleted records, but I'll make this "fuse" here, as a best practice.

Reply:
Actually, no. We need to `add` the `deleted` property to the state for the particular stream,  not emit the state for deleted records separately, in this case, we will override the actual state with the `deleted` state instead. I'm leaving the logic as is - works well.

Reply:
ok 👍🏼